common/src/main/java/com/wynntils/core/notifications/NotificationManager.java

```diff
@@ -10,12 +10,26 @@
 import com.wynntils.gui.render.TextRenderTask;
 import com.wynntils.mc.utils.ComponentUtils;
 import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.Pair;
+import com.wynntils.utils.objects.TimedSet;
 import com.wynntils.wynn.event.NotificationEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
 import com.wynntils.wynn.utils.WynnUtils;
+import java.util.concurrent.TimeUnit;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public final class NotificationManager {
+    private static final TimedSet<Pair<String, MessageContainer>> cachedMessageSet =
```

Comment:
This Set could contain only a `MessageContainer`. If you make a method, `getOriginalMessage`, in MessageContainer, you can use it to compare messages here, and don't need to store it in the set.

Note that, This change is only to make this code a bit shorter/cleaner, it does not save that much memory.